pkg/apis/admissionregistration/types.go

```diff
@@ -123,6 +123,25 @@ type ValidatingAdmissionPolicy struct {
 	metav1.ObjectMeta
 	// Specification of the desired behavior of the ValidatingAdmissionPolicy.
 	Spec ValidatingAdmissionPolicySpec
+	// The status of the ValidatingAdmissionPolicy, including warnings that are useful to determine if the policy
+	// behaves in the expected way.
+	Status ValidatingAdmissionPolicyStatus `protobuf:"bytes,3,opt,name=status"`
+}
+
+// ValidatingAdmissionPolicyStatus represents the status of a ValidatingAdmissionPolicy.
+type ValidatingAdmissionPolicyStatus struct {
+	// ExpressionWarnings contains helpful information for expressions that can aid troubleshooting.
+	// +optional
+	ExpressionWarnings []ExpressionWarning `protobuf:"bytes,1,rep,name=expressionWarnings"`
+}
+
+// ExpressionWarning is a warning information that targets a specific expression.
+type ExpressionWarning struct {
```

Comment:
What should the `listType` be fore this? If it is `listType=map` then we define the key to be both fields so that multiple warnings are allowed per expression? (I can imagine arguments for `listType=atomic` as well)

Reply:
Oops yes. Fixing.

Reply:
I would vote for `+listType=atomic` but not map. The list annotations should match these of `Validations`
```go
	// Validations contain CEL expressions which is used to apply the validation.
	// A minimum of one validation is required for a policy definition.
	// +listType=atomic
	// Required.
	Validations []Validation `json:"validations" protobuf:"bytes,3,rep,name=validations"`
```